.github/workflows/integration_test.yaml

```diff
@@ -187,6 +201,12 @@ jobs:
         with:
           name: ${{ matrix.image }}
           path: /tmp/${{ matrix.image }}.tar
+      - name: Run Github Trivy Image Action
+        if: ${{ github.event.pull_request.head.repo.fork }}
```

Comment:
Is there any reason for skipping the image analysis when running from a fork?

Reply:
I'll have to check if it works accessing the registry. Would be something like this:

```
      - name: Log in to the Container registry
        if: ${{ !github.event.pull_request.head.repo.fork }}
        uses: docker/login-action@v2
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
      - name: Build and push image
        uses: docker/build-push-action@v3
        if: ${{ !github.event.pull_request.head.repo.fork }}
        with:
          context: .
          push: true
          tags: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ matrix.image }}:${{ github.run_id }}
          file: ${{ inputs.working-directory }}/${{ matrix.image }}.Dockerfile
      - name: Run Github Trivy Image Action
        uses: aquasecurity/trivy-action@master
        with:
          image-ref: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ matrix.image }}:${{ github.run_id }}
          trivy-config: ${{ inputs.trivy-image-config }}
        env:
          TRIVY_USERNAME: ${{ github.actor }}
          TRIVY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
The tarball should be present regardless of the registry. Later it is pushed to the GH registry or to the microk8s one, but at this point that shouldn't make any difference

Reply:
For forks you'd probably have to use the tarball generated in the "Build image as tarball" step

Reply:
Maybe I misunderstood. 
```
      - name: Build image as tarball
        uses: docker/build-push-action@v3
        if: ${{ github.event.pull_request.head.repo.fork }}
```
I thought that the tarball would be built only in this case of this if. I changed that because the tar was not found here:
https://github.com/canonical/operator-workflows/actions/runs/3960365804/jobs/6784446185#step:9:16

Reply:
I'll add there too.